perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace.java

```diff
@@ -73,7 +75,7 @@ public void checkAttributeSyntax(PerunSessionImpl sess, User user, Attribute att
 	@Override
 	public void checkAttributeSemantics(PerunSessionImpl sess, User user, Attribute attribute) throws WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
 		String userLogin = attribute.valueAsString();
-		if (userLogin == null) throw new WrongReferenceAttributeValueException(attribute, null, user, null, "Value can't be null");
+		if (userLogin == null && !allowNoLogin) throw new WrongReferenceAttributeValueException(attribute, null, user, null, "Value can't be null");
 
 		// Get all users who have set attribute urn:perun:member:attribute-def:def:login-namespace:[login-namespace], with the value.
 		List<User> usersWithSameLogin = sess.getPerunBl().getUsersManagerBl().getUsersByAttribute(sess, attribute);
```

Comment:
I am not sure how the `getUsersByAttribute` and `checkReservedLogins` behave for null login but I think the latter would throw `InternalErrorException`. I think we could skip all these checks in this case, what do you think?

Reply:
I think the checks could be skipped for null value, then there is no need for setting `allowNoLogin ` and rather check it before calling the superclass method.